frontend/app/src/components/account-management/CreateAccount.vue

```diff
@@ -258,145 +258,169 @@
+  components: { ExternalLink, RevealableInput, PremiumCredentials },
+  props: {
+    displayed: { required: true, type: Boolean },
+    loading: { required: true, type: Boolean },
+    error: { required: false, type: String, default: '' }
+  },
+  setup(props, { emit }) {
+    const { displayed, error } = toRefs(props);
 
-  newUser!: boolean;
-  username: string = '';
-  password: string = '';
-  passwordConfirm: string = '';
+    const { newUser } = setupNewUser();
+    const username: Ref<string> = ref('');
+    const password: Ref<string> = ref('');
+    const passwordConfirm: Ref<string> = ref('');
 
-  premiumEnabled: boolean = false;
-  userPrompted: boolean = false;
+    const premiumEnabled: Ref<boolean> = ref(false);
+    const userPrompted: Ref<boolean> = ref(false);
 
-  submitUsageAnalytics: boolean = true;
-  apiKey: string = '';
-  apiSecret: string = '';
-  syncDatabase: boolean = false;
+    const submitUsageAnalytics: Ref<boolean> = ref(true);
+    const apiKey: Ref<string> = ref('');
+    const apiSecret: Ref<string> = ref('');
+    const syncDatabase: Ref<boolean> = ref(false);
 
-  premiumFormValid: boolean = true;
-  credentialsFormValid: boolean = false;
-  errorMessages: string[] = [];
-  step = 1;
+    const premiumFormValid: Ref<boolean> = ref(true);
+    const credentialsFormValid: Ref<boolean> = ref(false);
+    const errorMessages: Ref<string[]> = ref([]);
+    const step: Ref<number> = ref(1);
 
-  readonly usernameRules = [
-    (v: string) =>
-      !!v ||
-      this.$t(
-        'create_account.select_credentials.validation.non_empty_username'
-      ),
-    (v: string) =>
-      (v && /^[0-9a-zA-Z_.-]+$/.test(v)) ||
-      this.$t('create_account.select_credentials.validation.valid_username')
-  ];
+    const form: Ref<any> = ref(null);
 
-  readonly passwordRules = [
-    (v: string) =>
-      !!v ||
-      this.$t('create_account.select_credentials.validation.non_empty_password')
-  ];
-  readonly passwordConfirmRules = [
-    (v: string) =>
-      !!v ||
-      this.$t(
-        'create_account.select_credentials.validation.non_empty_password_confirmation'
-      )
-  ];
+    const updateConfirmationError = () => {
+      if (errorMessages.value.length > 0) {
+        return;
+      }
 
-  private updateConfirmationError() {
-    if (this.errorMessages.length > 0) {
-      return;
-    }
-    this.errorMessages.push(
-      this.$t(
-        'create_account.select_credentials.validation.password_confirmation_missmatch'
-      ).toString()
-    );
-  }
+      errorMessages.value.push(
+        i18n
+          .t(
+            'create_account.select_credentials.validation.password_confirmation_mismatch'
+          )
+          .toString()
+      );
+    };
 
-  @Watch('password')
-  onPasswordChange() {
-    if (this.password && this.password !== this.passwordConfirm) {
-      this.updateConfirmationError();
-    } else {
-      this.errorMessages.pop();
-    }
-  }
+    watch(password, () => {
```

Comment:
the current value of password will be the first argument of the method, you can use that to avoid unreferencing furthermore in this case since the logic is quite similar we could combine the two watchers into one https://v3.vuejs.org/api/computed-watch-api.html#watching-multiple-sources